src/pages/components/number-input/usage.mdx

```diff
@@ -96,24 +131,177 @@ or decrease an incremental value.
-- Use sentence-style capitalization (only the first word in a phrase and any
-  proper nouns capitalized) for the label.
-- See further guidance for using number labels within
-  [forms](/components/form/usage).
+### Anatomy
+
+<Row>
+<Column colLg={8}>
+
+![number input anatomy](images/number-input-usage-2.png)
+
+</Column>
+</Row>
+
+1. **Label**: Text that informs the user about the content they need to enter in
+   the field. It is required unless you get an approved accessibility exemption.
+2. **Numeric value**: The value changes when users entered into the field.
+3. **Helper text** (optional, except for error and warning): Assistive text that
+   can provide additional aid or context to the user. Often used to explain the
+   correct data format.
+4. **Field**: The container in which a user enters data.
+5. **Subtract icon**: This icon triggers decremental values.
+6. **Add icon**: This icon triggers incremental values.
+7. **Status icon**: Indicates the state of the number input, either error or
+   warning.
+
+### Sizing
+
+Number input has three sizes in height: **small**, **medium**, and **large**.
+The width varies in sizes based on content, layout, and design.
+
+| Size        | Height (px/rem) | Use case                                                                                                                                              |
+| ----------- | --------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Small (sm)  | 32 / 2          | Use when space is constricted or when placing a number input in a form that is long and complex.                                                      |
+| Medium (md) | 40 / 2.5        | This is the default size and the most commonly used size. When in doubt, use the medium size.                                                         |
+| Large (lg)  | 48 / 3          | Use when there is a lot of space to work with. The large size is typically used in simple forms or when a number input is placed by itself on a page. |
+
+<Row>
+<Column colLg={8}>
+
+![number input sizes](images/number-input-usage-3.png)
+
+</Column>
+</Row>
+
+## Content
+
+### Main elements
+
+#### Label
+
+- Always include a label so that users know what to complete the field
+- Keep the label short and concise
+- Use sentence-style capitalization for the label
+
+<DoDontRow>
+  <DoDont caption="Do include a label">
+
+![Do include a label](images/number-input-usage-4.png)
+
+  </DoDont>
+  <DoDont type="dont" caption="Do not hide label that helps user understand the context of number input">
+
+![Do not hide label](images/number-input-usage-5.png)
+
+  </DoDont>
+</DoDontRow>
+
+#### Helper text
+
+- Indicate the input value if it has a maximum or minimum
+- Helper text is an optional, but required for warning and error states
+
+### Overflow content
```

Comment:
I don't think you need this specific section unless there is an overflow behavior. Otherwise just move this content below each section in main elements.